### PR TITLE
Replace cookie consent with micro:bit version

### PR DESCRIPTION
**Pull request by microbit-matt-hillsdon**
This is common to all Foundation apps.

I've slimmed down the CookieManager to just be responsible for cookie read/write rather than consent.

I've deleted the cookie UI we no longer use. Although likely to cause conflicts in future, these will at least be easy to address by retaining the delete. Otherwise I had to retain stub functionality in CookieManager for the unused UI.

All non-analytics scenarios are likely to be classed as necessary/essential so this seems fairly straightforward.

### CODE REVIEW

File: src/script/stores/complianceStore.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  w.commonConsent?.show({ userTriggered, config });
+}
+
+export function manageCookies() {
+  showConsent({ userTriggered: true });
+}
+
+export const consent = writable<CookieConsent | undefined>(undefined);
+
+const w = window as any;
```
Comment (microbit-matt-hillsdon) [inline src/script/stores/complianceStore.ts:47]: Not sure this is necessary here. It was just a hack around the linting in the project I copied this from as our common consent API is untyped.

File: src/script/stores/complianceStore.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      : undefined,
+  custom: [
+    {
+      type: 'cookie',
+      category: 'essential',
+      name: 'something',
+      domain: 'ml-tool.microbit.org',
+      purpose: 'Something helpful here',
+    },
+    // Some of the Svelte stores use local storage, this needs investigating
```
Comment (microbit-matt-hillsdon) [inline src/script/stores/complianceStore.ts:30]: Needs filling in.

Comment (microbit-matt-hillsdon) [reply src/script/stores/complianceStore.ts]: d00b75009dd8e584c553216faea82ba59219cbf4

File: src/script/stores/complianceStore.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+};
+
+function showConsent(
+  { userTriggered }: { userTriggered: boolean } = { userTriggered: false },
+) {
+  const w = window as any;
+  w.commonConsent?.show({ userTriggered, config });
+}
+
+export function manageCookies() {
```
Comment (microbit-matt-hillsdon) [inline src/script/stores/complianceStore.ts:63]: This is unused for the moment but there needs to be a way to revoke consent which can be done via the cookie dialog. We should wire this up to a new menu similar to other Foundation apps with other compliance items. We'll leave it unused for now and tackle this in a future PR.

**Comment by micque01** [comment]
This seems to be working nicely, it also fixes a problem we were having with cookie dialog flash on page reload 😄 

Is there any more work that needs doing on this before it can come out of draft and we can get it merged?

**Pull request merged**